client/src/components/WritePost/index.jsx

```diff
@@ -0,0 +1,131 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import TextareaAutosize from 'react-autosize-textarea';
+import {
+  Button, Radio, Divider, Message,
+} from 'semantic-ui-react';
+
+import { createPost } from '../../API';
+import styles from './WritePost.module.css';
+import { useBoolean } from '../../hooks';
+
+const WritePost = ({
+  title, content, category, onTitleChange, onContentChange,
+}) => {
+  const [isPublic, setIsPublic] = useBoolean(true);
+  const [isSaveLoading, setIsSaveLoading] = useState(false);
+  const [isPublishedLoading, setIsPublishedLoading] = useState(false);
+
+  const handlePublish = async () => {
+    setIsPublishedLoading(true);
+    const body = {
+      author: '5d5d60a51930b06b21e65765',
+      title,
+      content,
+      category,
+      public: isPublic,
+      published: true,
+    };
+    await createPost(body);
+    setIsPublishedLoading(false);
+  };
+
+  const handleSave = async () => {
+    setIsSaveLoading(true);
+    const body = {
+      author: '5d5d60a51930b06b21e65765',
+      title,
+      content,
+      public: isPublic,
+      published: false,
+    };
+    await createPost(body);
```

Comment:
try/catch ?

Reply:
`createPost` have a try/catch inside it. Should I wrap function call in a different try/catch as well?